package/csac_daemon/src/csac_daemon.c

```diff
@@ -0,0 +1,307 @@
+
+#define SBP_FRAMING_MAX_PAYLOAD_SIZE (255u)
+#define CSAC_TELEM_UPDATE_INTERVAL (1000u)
+#define HEADER_SEND_DIVISOR (10)
+#define POLL_SLEEP_MICROSECONDS (100000)
+
+u8 *port_name = NULL;
+u8 *command_string = NULL;
+bool csac_telemetry_enabled = false;
+
+struct csac_ctx_s {
+  sbp_pubsub_ctx_t *sbp_ctx;
+  serial_port_t *port;
+  char csac_telem[SBP_FRAMING_MAX_PAYLOAD_SIZE];
+  char csac_header[SBP_FRAMING_MAX_PAYLOAD_SIZE];
+};
+
+static bool csac_daemon_enabled()
+{
+  return csac_telemetry_enabled;
+}
+
+static void usage(char *command)
+{
+  printf("Usage: %s\n", command);
+
+  puts("\nMain options");
+  puts("\t--serial-port <serial port for csac daemon>");
+}
+
+static int parse_options(int argc, char *argv[])
+{
+  enum { OPT_ID_SERIAL_PORT = 1, OPT_ID_AT_COMMAND = 2 };
+
+  const struct option long_opts[] = {
+    {"serial-port", required_argument, 0, OPT_ID_SERIAL_PORT},
+    {0, 0, 0, 0},
+  };
+
+  int opt;
+  while ((opt = getopt_long(argc, argv, "", long_opts, NULL)) != -1) {
+    switch (opt) {
+    case OPT_ID_SERIAL_PORT: {
+      port_name = (u8 *)optarg;
+    } break;
+
+    default: {
+      puts("Invalid option");
+      return -1;
+    } break;
+    }
+  }
+
+  if (port_name == NULL) {
+    puts("Missing port");
+    return -1;
+  }
+
+  return 0;
+}
+
+static int read_until_cr_lf(int fd, char *outbuf, size_t maxlen)
+{
+  int nbytes;   /* Number of bytes read */
+  char *bufptr; /* Current char in buffer */
+  bufptr = outbuf;
+  char *eob = outbuf + maxlen;
+  while ((nbytes = read(fd, bufptr, (size_t)(eob - bufptr) - 1)) > 0) {
+    bufptr += nbytes;
+    if ((size_t)(bufptr - outbuf) > 2) {
+      if (bufptr[-1] == '\n' || bufptr[-1] == '\r') {
+        break;
+      }
+    }
+  }
+  *bufptr = '\0';
+  return (bufptr - outbuf);
+}
+
+/**
+ * @brief send_csac_telem
+ *
+ * Sends csac telemetry string as SBP message to external endpoint
+ * @param csac_telem: pointer to null_terminated c string
+ */
+static void send_csac_telem(struct csac_ctx_s *ctx)
+{
+  static int count;
+  size_t telem_string_len = 0;
+  telem_string_len = strnlen(ctx->csac_telem, 254);
+  msg_csac_telemetry_t *msg = alloca(SBP_FRAMING_MAX_PAYLOAD_SIZE);
+  memcpy(msg->telemetry, ctx->csac_telem, telem_string_len);
+  msg->id = 0;
+  msg->telemetry[telem_string_len] = '\0'; /* Ensure null termination */
+  sbp_tx_send(sbp_pubsub_tx_ctx_get(ctx->sbp_ctx),
+              SBP_MSG_CSAC_TELEMETRY,
+              (u8)(0xFF & (telem_string_len + 1)),
+              (u8 *)msg);
+  count++;
+  if ((count % HEADER_SEND_DIVISOR) == 0) {
+    memcpy(msg->telemetry, ctx->csac_header, telem_string_len);
+    telem_string_len = strnlen(ctx->csac_header, 254);
+    msg->telemetry[telem_string_len] = '\0'; /* Ensure null termination */
+    sbp_tx_send(sbp_pubsub_tx_ctx_get(ctx->sbp_ctx),
+                SBP_MSG_CSAC_TELEMETRY_LABELS,
+                (u8)(0xFF & (telem_string_len + 1)),
```

Comment:
Why `(0xFF & `?  If it's a form of max, just use the SWFT_MAX macro to be more explicit.